File: Data Structures & Algorithms/is-anagram/submission-10.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> count;
        for (char c : s) {
            if (count.contains(c)) {
                count[c] += 1;
            } else {
                count[c] = 1;
            }
        }

        for (char c : t) {
            if (count.contains(c)) {
                count[c] -= 1;
            } else {
                count[c] = 1;
            }

        }

        for (const auto& [c, freq] : count) {
            if (freq != 0) {
                return false;
            }
        }
        return true;

    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-10.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = Counter(nums)

        heap = []
        for (num, freq) in count.items():
            heapq.heappush(heap, (freq, num))
            if len(heap) > k:
                heapq.heappop(heap)
        
        ans = []
        for _ in range(k):
            ans.append(heapq.heappop(heap)[1])
        
        return ans